shared/bls/bls.go

```diff
@@ -4,205 +4,248 @@
 package bls
 
 import (
+	"crypto/rand"
 	"encoding/binary"
 	"io"
-	"time"
+	"math/big"
 
-	"github.com/karlseguin/ccache"
-	g1 "github.com/phoreproject/bls/g1pubs"
 	"github.com/pkg/errors"
 	"github.com/prysmaticlabs/prysm/shared/bytesutil"
 	"github.com/prysmaticlabs/prysm/shared/featureconfig"
-)
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
 
-var pubkeyCache = ccache.New(ccache.Configure())
+	bls12 "github.com/kilic/bls12-381"
+)
 
 // CurveOrder for the BLS12-381 curve.
 const CurveOrder = "52435875175126190479447740508185965837690552500527637822603658699938581184513"
 
+var curveOrder, _ = new(big.Int).SetString(CurveOrder, 10)
+
 // Signature used in the BLS signature scheme.
 type Signature struct {
-	val *g1.Signature
-}
-
-// SecretKey used in the BLS signature scheme.
-type SecretKey struct {
-	val *g1.SecretKey
+	s *bls12.PointG2
 }
 
 // PublicKey used in the BLS signature scheme.
 type PublicKey struct {
-	val *g1.PublicKey
+	p *bls12.PointG1
+}
+
+// SecretKey used in the BLS signature scheme.
+type SecretKey struct {
+	p *big.Int
 }
 
 // RandKey creates a new private key using a random method provided as an io.Reader.
 func RandKey(r io.Reader) (*SecretKey, error) {
-	k, err := g1.RandKey(r)
+	k, err := rand.Int(r, curveOrder)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not initialize secret key")
+		return nil, err
 	}
-	return &SecretKey{val: k}, nil
+	return &SecretKey{k}, nil
 }
 
 // SecretKeyFromBytes creates a BLS private key from a LittleEndian byte slice.
 func SecretKeyFromBytes(priv []byte) (*SecretKey, error) {
-	k := bytesutil.ToBytes32(priv)
-	val := g1.DeserializeSecretKey(k)
-	if val.GetFRElement() == nil {
+	b := bytesutil.ToBytes32(priv)
+	k := new(big.Int).SetBytes(b[:])
+	if curveOrder.Cmp(k) < 0 {
 		return nil, errors.New("invalid private key")
 	}
-	return &SecretKey{val}, nil
+	return &SecretKey{p: k}, nil
 }
 
 // PublicKeyFromBytes creates a BLS public key from a  LittleEndian byte slice.
 func PublicKeyFromBytes(pub []byte) (*PublicKey, error) {
 	if featureconfig.FeatureConfig().SkipBLSVerify {
 		return &PublicKey{}, nil
 	}
-	cv := pubkeyCache.Get(string(pub))
```

Comment:
Why remove this? This method was called in high frequency with infrequently changing parameters. Aligning the bytes to the curve was expensive.